### PR TITLE
VLASS ingestion

### PR DESCRIPTION
**Pull request by mcoughlin**
This adds a script to enable VLASS ingestion into kowalski.

### CODE REVIEW

File: tools/ingest_vlass.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    # 2D position on the sphere, ID:
+    m.db[catalog_name].create_index(
+        [("coordinates.radec_geojson", "2dsphere"), ("_id", 1)], background=True
+    )
+
+    params = [(f, catalog_name, args.batch_size, args.rm, args.v) for f in files]
+
+    with Pool(processes=args.n_processes) as pool:
+        # list+pool.imap is a trick to make tqdm work here
+        list(tqdm(pool.imap(process_file, params), total=len(files)))
```
Comment (dmitryduev) [inline tools/ingest_vlass.py:176]: `pool.imap+tqdm` is broken, please use @kmshin1397's `istarmap` instead, see https://github.com/dmitryduev/kowalski/blob/master/tools/ingest_ztf_source_features.py#L13

File: tools/ingest_vlass.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    # clean up:
+    if rm:
+        os.remove(file)
+        if verbose:
+            log(f"Successfully removed {file}")
+
+
+if __name__ == "__main__":
+    # Note CSV data file comes from https://cirada.ca/vlasscatalogueql0
+    parser = argparse.ArgumentParser()
```
Comment (dmitryduev) [inline tools/ingest_vlass.py:124]: Could you please scrap all the `argparse` stuff replace it with `python-fire`? (refactor the logic below into a `run()` function and use `fire.Fire(run)`). See https://github.com/dmitryduev/kowalski/blob/master/tools/ingest_ztf_source_features.py for an example.

File: tools/ingest_vlass.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+    # clean up:
+    if rm:
+        os.remove(file)
+        if verbose:
+            log(f"Successfully removed {file}")
+
+
+if __name__ == "__main__":
+    # Note CSV data file comes from https://cirada.ca/vlasscatalogueql0
```
Comment (dmitryduev) [inline tools/ingest_vlass.py:123]: How big are those? Maybe mention the total unpacked size here?

**Review by dmitryduev — CHANGES_REQUESTED** [review]
LGTM, just a couple relatively minor requests.

**Comment by dmitryduev** [comment]
@mcoughlin actually one more request if you don't mind: could you please write a test similar to one in https://github.com/dmitryduev/kowalski/blob/master/tests/test_tools.py#L33? I.e. put a tiny example in /data and test ingestion.

File: tools/ingest_vlass.py [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    deg2dms,
+    deg2hms,
+    load_config,
+    log,
+    Mongo,
+)
+
+""" load config and secrets """
+config = load_config(config_file="config.yaml")["kowalski"]
+
```
Comment (dmitryduev) [inline tools/ingest_vlass.py:22]:
```suggestion
# init db if necessary
init_db_sync(config=config)
```

File: tools/ingest_vlass.py [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+import argparse
+from multiprocessing.pool import Pool
+import pandas as pd
+import pathlib
+import os
+from tqdm.auto import tqdm
+
+from utils import (
+    deg2dms,
+    deg2hms,
```
Comment (dmitryduev) [inline tools/ingest_vlass.py:14]:
```suggestion
    deg2hms,
    init_db_sync,
```

**Review by dmitryduev — APPROVED** [review]
Excellente! 🐧

File: tools/ingest_vlass.py [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+    # disconnect from db:
+    try:
+        mongo.client.close()
+    finally:
+        log("Successfully disconnected from db")
+
+
+def run(
+    path: str = "./",
+    num_processes: int = mp.cpu_count(),
```
Comment (dmitryduev) [inline tools/ingest_vlass.py:116]: Noticed that there's only one file, so this should default to 1.

Comment (mcoughlin) [reply tools/ingest_vlass.py]: Yup fixed!

**Comment by dmitryduev** [comment]
@mcoughlin do you want to x-match the alert stream against VLASS?
btw, I've already ingested the catalog ;)

**Comment by mcoughlin** [comment]
@dmitryduev My intuition is that this is *most* useful for our light curve work (and in particular the Fermi searches...), but we should definitely discuss within SCoPe.

**Review by kmshin1397 — APPROVED** [review]
Looks good @mcoughlin! Are these VLASS files ever very large? If you foresee needing to ingest a lot more of these at some point, I could imagine we'd be able to speed things up significantly by processing multiple chunks of the `read_csv()` loop at a time with multiprocessing as well as (or instead of) a process per file, especially if you have a small number of very large files. If that's not really a concern then I think this is fine to go in as is :)

**Comment by mcoughlin** [comment]
@kmshin1397 Thanks! Files are a few GB and tend to go in quickly, but maybe we do that for the next catalog for practice?

**Pull request merged**